### ISSUE TITLE
AvvioError: Plugin must be a function or a promise when run using fastify-cli

### ISSUE THREAD

**Issue by sujeet-agrahari** [fastify/fastify#4946]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

4.21.0

### Plugin version

4.5.1

### Node.js version

20

### Operating system

macOS

### Operating system version (i.e. 20.04, 11.3, 10)

13.5

### Description

```
> node-fastify-architecture@1.0.0 start /Users/sujeet/github/node-fastify-architecture
> fastify start -l info ./src/server.js

AvvioError [Error]: Plugin must be a function or a promise. Received: 'object'
    at assertPlugin (/Users/sujeet/github/node-fastify-architecture/node_modules/.pnpm/avvio@8.2.1/node_modules/avvio/boot.js:207:11)
    at Boot._addPlugin (/Users/sujeet/github/node-fastify-architecture/node_modules/.pnpm/avvio@8.2.1/node_modules/avvio/boot.js:240:12)
    at Boot.use (/Users/sujeet/github/node-fastify-architecture/node_modules/.pnpm/avvio@8.2.1/node_modules/avvio/boot.js:216:25)
    at server.<computed> [as register] (/Users/sujeet/github/node-fastify-architecture/node_modules/.pnpm/avvio@8.2.1/node_modules/avvio/boot.js:40:14)
    at runFastify (/Users/sujeet/github/node-fastify-architecture/node_modules/.pnpm/fastify-cli@5.8.0/node_modules/fastify-cli/start.js:149:17)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  code: 'AVV_ERR_PLUGIN_NOT_VALID'
}
[13:03:36.034] INFO (97931): 
 └── /
    ├── fastify-overview-ui/
    │   └── * (HEAD, GET)
    ├── json-overview-ui (GET, HEAD)
    └── api/v1/auth/login (POST)

[13:03:36.035] INFO (97931): Server listening at http://127.0.0.1:3000
 ELIFECYCLE  Command failed with exit code 1.
```

### Steps to Reproduce

```
import Fastify from 'fastify';
import autoload from '@fastify/autoload';
import { join } from 'desm';
import { errorHandler } from './middlewares/error-handler.js';
import { notFoundHandler } from './middlewares/not-found-error.js';
import fastifyOverview from 'fastify-overview';

export default async function buildApp(opts = {}) {
  const app = Fastify(opts);

  await app.register(fastifyOverview, { addSource: true }); // -> here happens when this is loaded
  
  app.register(autoload, {
    dir: join(import.meta.url, 'plugins'),
    options: Object.assign({}, opts)
  })

  // auto load modules
  await app.register(autoload, {
    dir: join(import.meta.url, 'modules'),
    maxDepth: 1,
    dirNameRoutePrefix: false,
    matchFilter: (path) => path.endsWith('.module.js'),
    options: Object.assign({ prefix: '/api/v1' }, opts)
  });

  app.setErrorHandler(errorHandler);

  app.setNotFoundHandler(notFoundHandler);


  return app;
}

```

### Expected Behavior

It shouldn't fail!

- It throws error when run using `fastify-cli`, see the ran command in above
```
fastify start -l info ./src/server.js

```
- It works when ran using,
```
nodemon --require=dotenv/config -L src/server.js
```
Below are the dependencies with their versions:

```
  "dependencies": {
    "@fastify/autoload": "^5.7.1",
    "@fastify/sensible": "^5.2.0",
    "@prisma/client": "^5.0.0",
    "bcrypt": "^5.1.0",
    "close-with-grace": "^1.2.0",
    "config": "^3.3.9",
    "desm": "^1.3.0",
    "dotenv": "^16.3.1",
    "fastify": "^4.21.0",
    "fastify-overview": "^3.3.2",
    "fastify-overview-ui": "^2.2.1",
    "fastify-plugin": "^4.5.1",
    "js-yaml": "^4.1.0",
    "jsonwebtoken": "^9.0.1",
    "nodemon": "^3.0.1"
  },
```

**Comment by mcollina** [comment]
Thanks for reporting!

Can you provide steps to reproduce? We often need a [reproducible example](https://stackoverflow.com/help/minimal-reproducible-example), e.g. some code that allows someone else to recreate your problem by just copying and pasting it. If it involves more than a couple of different file, create a new repository on GitHub and add a link to that.

**Comment by Eomm** [comment]
could you print the `fastifyOverview` import from `import fastifyOverview from 'fastify-overview';`?

I think I did not export the `default` one.

This should work:

```
import * as fastifyOverview from 'fastify-overview'
```

**Comment by sujeet-agrahari** [comment]
Hi @Eomm @mcollina The below solution didn't work.
Below is the minimal reproducible example.
**app.js**
```js
import Fastify from 'fastify';
import fastifyOverview from 'fastify-overview'

export default async function buildApp(opts) {
  const app = Fastify(opts);
  console.log(fastifyOverview);

  await app.register(fastifyOverview, { addSource: true }); // -> here happens when this is loaded
  return app;
}
```

**server.js**
```
import buildApp from "./app.js";
const port = 3000;
const host = '127.0.0.1';

const app = await buildApp({
  logger: true
})

await app.listen({ port, host });
```
**package.json**
```
{
  "name": "fastify",
  "version": "1.0.0",
  "description": "This project was bootstrapped with Fastify-CLI.",
  "main": "app.js",
  "type": "module",
  "directories": {
    "test": "test"
  },
  "scripts": {
    "start": "fastify start -l info server.js",
    "dev": "fastify start -w -l info -P server.js"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "dependencies": {
    "fastify": "^4.0.0",
    "fastify-cli": "^5.8.0",
    "fastify-overview": "^3.3.2"
  }
}
```

**Comment by Eomm** [comment]
I released 3.4.0 yesterday, could you retry with a fresh installation?

Anyway it is not a fastify issue, but a plugin one

**Comment by Eomm** [comment]
Oh gosh, I did not see you were creating the `const app = Fastify(opts);`!

yes, it is not needed with fastify-cli: it creates the server and registers your application as a plugin
